packages/proxy/lib/http/util/service-worker-injector.ts

```diff
@@ -0,0 +1,228 @@
+    self.removeEventListener = (type: string, listener: EventListenerOrEventListenerObject, options?: boolean | AddEventListenerOptions) => {
+      if (type === 'fetch' && isValidListener(listener)) {
+        // get the capture value so we know which map to remove the listener from
+        const capture = getCaptureValue(options)
+        const newListener = capture ? _captureListenersMap.get(listener) : _nonCaptureListenersMap.get(listener)
+
+        capture ? _captureListenersMap.delete(listener) : _nonCaptureListenersMap.delete(listener)
+        _listenerCount--
+
+        // If the listener is an object with a handleEvent method, we need to remove the wrapped function
+        if (typeof listener === 'object' && typeof listener.handleEvent === 'function') {
+          _targetToWrappedHandleEventMap.delete(listener)
+          _targetToOrigHandleEventMap.delete(listener)
+        }
+
+        __cypressServiceWorkerSendHasFetchEventHandlers()
+
+        return oldRemoveEventListener(type, newListener!, options)
+      }
+
+      return oldRemoveEventListener(type, listener, options)
+    }
+
+    const originalPropertyDescriptor = Object.getOwnPropertyDescriptor(
+      self,
+      'onfetch',
+    )
+
+    if (!originalPropertyDescriptor) {
+      return
+    }
+
+    // Overwrite the onfetch property so we can
+    // determine if the service worker handled the request
+    Object.defineProperty(
+      self,
+      'onfetch',
+      {
+        configurable: originalPropertyDescriptor.configurable,
+        enumerable: originalPropertyDescriptor.enumerable,
+        get () {
+          return originalPropertyDescriptor.get?.call(this)
+        },
+        set (value: (event) => void) {
+          let newHandler
+
+          if (value) {
+            newHandler = __cypressWrapListener(value)
+          }
+
+          originalPropertyDescriptor.set?.call(this, newHandler)
+
+          __cypressServiceWorkerSendHasFetchEventHandlers()
+        },
+      },
+    )
+
+    // listen for the activate event so we can inform the
+    // backend whether or not the service worker has a handler
+    self.addEventListener('activate', () => {
+      __cypressServiceWorkerSendHasFetchEventHandlers()
+
+      // if the binding has not been created yet, we need to wait for it
+      if (!self.__cypressServiceWorkerClientEvent) {
+        const timer = setInterval(() => {
+          if (self.__cypressServiceWorkerClientEvent) {
+            clearInterval(timer)
+
+            // send any events that were queued
+            _eventQueue.forEach((event) => {
+              self.__cypressServiceWorkerClientEvent(JSON.stringify(event))
+            })
+
+            _eventQueue = []
+          }
+        }, 5)
+      }
+    })
+  }
+
+  const updatedBody = `
+let __cypressScriptEvaluated = false;
+(${__cypressInjectIntoServiceWorker})();
+${body};
+__cypressScriptEvaluated = true;`
```

Comment:
Why is this set here rather than after line 223?

Reply:
Should we rename it maybe then? Serviceworker script evaluated or something?

Reply:
I prefixed it with `__cypress` to ensure we don't collide with anything in the user's script.

Reply:
renamed to `__cypressIsScriptEvaluated`. [8378c36](https://github.com/cypress-io/cypress/pull/28900/commits/8378c367e06f0fc50b09c29bc67263d88e5dde5b)